test/tic_tac_toe/ai_test.clj

```diff
@@ -192,4 +194,49 @@
   (let [player  (->AiPlayer "o")
         game { :board [["_" "_" "_" ]["_" "_" "_" ]["_" "_" "_" ]] :ai-marker "o" :player-marker "x"}]
     (testing "creates defrecord of player protocol"
-      (is (= (assoc game :board [["o" "_" "_"] ["_" "_" "_"] ["_" "_" "_"]]) (next-move player game))))))
\ No newline at end of file
+      (is (= (assoc game :board [["o" "_" "_"] ["_" "_" "_"] ["_" "_" "_"]]) (next-move player game))))))
+
+(defn win-or-draw? [player-marker winner-state]
+ (or (= player-marker winner-state) (= player-marker "its a draw")))
+
+(defn correct-ai-choice? [game]
+  (let[winner (winner? (:board game))]
+     (cond
+      (= winner (:ai-marker game)) true
+      (= winner "its a draw") true
+      (= winner nil) true
+      :else false )))
+
+(declare check-every-possible-gamestate)
+
+(defn check-gamestate [gamestate]
+  (let [winner-state (winner? (:board gamestate))]
+    (if (nil? winner-state)
+      (check-every-possible-gamestate gamestate)
+      (if (correct-ai-choice? gamestate)
```

Comment:
Does `correct-ai-choice?` return a boolean? If so, you shouldn't need this conditional.